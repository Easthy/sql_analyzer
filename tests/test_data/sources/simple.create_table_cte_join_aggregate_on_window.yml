source.user:
  - id
  - load_date
  - email
source.user_status:
  - user_id
  - status
source.user_payment:
  - user_id
  - operation_id
  - id