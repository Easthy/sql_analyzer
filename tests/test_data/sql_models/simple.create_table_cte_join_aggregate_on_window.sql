BEGIN TRANSACTION;

CREATE TABLE IF NOT EXISTS simple.create_table_cte_join_aggregate_on_window (
    user_id INTEGER,
    email VARCHAR,
    status VARCHAR,
    payment_cnt INTEGER ENCODE AZ64
)
    DISTSTYLE KEY
    DISTKEY(user_id)
    SORTKEY(status, user_id)
;

INSERT INTO simple.create_table_cte_join_aggregate_on_window
WITH _cte AS (
    SELECT user.id,
           user.email,
           ROW_NUMBER() OVER (
                  PARTITION BY user.id
                      ORDER BY user.load_date DESC
           ) AS rn
      FROM source.user
)
  SELECT _cte.id,
         _cte.email,
         user_status.status,
         COUNT(
             CASE
                  WHEN user_payment.operation_id IN (1, 2)
                       THEN user_payment.id
                  ELSE NULL
             END
         ) AS payment_cnt
    FROM _cte
  
         LEFT JOIN source.user_status
         ON user_status.user_id = _cte.id
  
         LEFT JOIN source.user_payment
         ON user_payment.user_id = _cte.id

   WHERE _cte.rn = 1
GROUP BY 1, 2
;

END TRANSACTION;