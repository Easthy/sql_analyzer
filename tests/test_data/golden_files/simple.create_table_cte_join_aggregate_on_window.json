{
  "nodes": [
    { "id": "col:simple.create_table_cte_join_aggregate_on_window.email" },
    { "id": "col:simple.create_table_cte_join_aggregate_on_window.payment_cnt" },
    { "id": "col:simple.create_table_cte_join_aggregate_on_window.status" },
    { "id": "col:simple.create_table_cte_join_aggregate_on_window.user_id" },
    { "id": "col:source.user.email" },
    { "id": "col:source.user.id" },
    { "id": "col:source.user.load_date" },
    { "id": "col:source.user_payment.id" },
    { "id": "col:source.user_payment.operation_id" },
    { "id": "col:source.user_payment.user_id" },
    { "id": "col:source.user_status.status" },
    { "id": "col:source.user_status.user_id" },
    { "id": "tbl:simple.create_table_cte_join_aggregate_on_window" },
    { "id": "tbl:source.user" },
    { "id": "tbl:source.user_payment"},
    { "id": "tbl:source.user_status" }
  ],
  "links": [
    {
      "source": "col:simple.create_table_cte_join_aggregate_on_window.email",
      "target": "col:source.user.email",
      "type": "column_dependency"
    },
    {
      "source": "col:simple.create_table_cte_join_aggregate_on_window.payment_cnt",
      "target": "col:source.user_payment.id",
      "type": "column_dependency"
    },
    {
      "source": "col:simple.create_table_cte_join_aggregate_on_window.payment_cnt",
      "target": "col:source.user_payment.operation_id",
      "type": "column_dependency"
    },
    {
      "source": "col:simple.create_table_cte_join_aggregate_on_window.status",
      "target": "col:source.user_status.status",
      "type": "column_dependency"
    },
    {
      "source": "col:simple.create_table_cte_join_aggregate_on_window.user_id",
      "target": "col:source.user.id",
      "type": "column_dependency"
    },
    {
      "source": "tbl:simple.create_table_cte_join_aggregate_on_window",
      "target": "col:simple.create_table_cte_join_aggregate_on_window.email",
      "type": "contains_column"
    },
    {
      "source": "tbl:simple.create_table_cte_join_aggregate_on_window",
      "target": "col:simple.create_table_cte_join_aggregate_on_window.payment_cnt",
      "type": "contains_column"
    },
    {
      "source": "tbl:simple.create_table_cte_join_aggregate_on_window",
      "target": "col:simple.create_table_cte_join_aggregate_on_window.status",
      "type": "contains_column"
    },
    {
      "source": "tbl:simple.create_table_cte_join_aggregate_on_window",
      "target": "col:simple.create_table_cte_join_aggregate_on_window.user_id",
      "type": "contains_column"
    },
    {
      "source": "tbl:simple.create_table_cte_join_aggregate_on_window",
      "target": "tbl:source.user",
      "type": "table_dependency"
    },
    {
      "source": "tbl:simple.create_table_cte_join_aggregate_on_window",
      "target": "tbl:source.user_payment",
      "type": "table_dependency"
    },
    {
      "source": "tbl:simple.create_table_cte_join_aggregate_on_window",
      "target": "tbl:source.user_status",
      "type": "table_dependency"
    },
    {
      "source": "tbl:source.user",
      "target": "col:source.user.email",
      "type": "contains_column"
    },
    {
      "source": "tbl:source.user",
      "target": "col:source.user.id",
      "type": "contains_column"
    },
    {
      "source": "tbl:source.user",
      "target": "col:source.user.load_date",
      "type": "contains_column"
    },
    {
      "source": "tbl:source.user_payment",
      "target": "col:source.user_payment.id",
      "type": "contains_column"
    },
    {
      "source": "tbl:source.user_payment",
      "target": "col:source.user_payment.operation_id",
      "type": "contains_column"
    },
    {
      "source": "tbl:source.user_payment",
      "target": "col:source.user_payment.user_id",
      "type": "contains_column"
    },
    {
      "source": "tbl:source.user_status",
      "target": "col:source.user_status.status",
      "type": "contains_column"
    },
    {
      "source": "tbl:source.user_status",
      "target": "col:source.user_status.user_id",
      "type": "contains_column"
    }
  ]
}